src/main/scala/scalafix/sbt/SemanticRuleValidator.scala

```diff
@@ -0,0 +1,67 @@
+package scalafix.sbt
+
+import java.nio.file.Path
+
+import sbt.{CrossVersion, ModuleID}
+import scalafix.interfaces.ScalafixArguments
+
+import scala.collection.mutable.ListBuffer
+
+private[sbt] class SemanticRuleValidator(ifNotFound: => SemanticdbNotFound) {
```

Comment:
Can we remove the accessibility restriction and by-name argument?
```suggestion
class SemanticRuleValidator(ifNotFound: SemanticdbNotFound) {
```

Reply:
Thank you for pointing out.
By-name argument can be removed since it's not so important (and not good for binary compat?).

IMO `private[sbt]` should be here, since it's not intended to be referenced by users. Otherwise, we may break users code due to changing this internally used class.
What do you think about this point?

Reply:
We can move this to the internal package if it's internal, my experience is that it's easier for users to see what is internal vs. public when internal code is moved to a separate package instead of adding `private[pkg]` everywhere

Reply:
https://github.com/scalacenter/sbt-scalafix/tree/master/src/main/scala/scalafix/internal/sbt

Reply:
Thank you for confirmation.
While I prefer restriction over convention, I can understand we don't want to mix up two different ways to say *"This is for internal!"*.

I will update this PR shortly.

Reply:
It's possible to access `private[pkg]` symbols by writing proxy methods in that package. Both approaches are leaky, but using `internal` in the package name at least makes it easier to see in the client implementation or stack traces when the user is accessing internal APIs.

Reply:
Thank you for throughout reviewing and merging all my pull requests!

> it easier to see in the client implementation or stack traces when the user is accessing internal APIs.

While I know there are *escape hatches* in both approaches, noticeability in stack traces is a fair point. I'm fine to go the `internal` approach. :)